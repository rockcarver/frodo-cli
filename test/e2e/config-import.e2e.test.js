/**
 * Follow this process to write e2e tests for the CLI project:
 *
 * 1. Test if all the necessary mocks for your tests already exist.
 *    In mock mode, run the command you want to test with the same arguments
 *    and parameters exactly as you want to test it, for example:
 *
 *    $ FRODO_MOCK=1 frodo conn save https://openam-frodo-dev.forgeblocks.com/am volker.scheuber@forgerock.com Sup3rS3cr3t!
 *
 *    If your command completes without errors and with the expected results,
 *    all the required mocks already exist and you are good to write your
 *    test and skip to step #4.
 *
 *    If, however, your command fails and you see errors like the one below,
 *    you know you need to record the mock responses first:
 *
 *    [Polly] [adapter:node-http] Recording for the following request is not found and `recordIfMissing` is `false`.
 *
 * 2. Record mock responses for your exact command.
 *    In mock record mode, run the command you want to test with the same arguments
 *    and parameters exactly as you want to test it, for example:
 *
 *    $ FRODO_MOCK=record frodo conn save https://openam-frodo-dev.forgeblocks.com/am volker.scheuber@forgerock.com Sup3rS3cr3t!
 *
 *    Wait until you see all the Polly instances (mock recording adapters) have
 *    shutdown before you try to run step #1 again.
 *    Messages like these indicate mock recording adapters shutting down:
 *
 *    Polly instance 'conn/4' stopping in 3s...
 *    Polly instance 'conn/4' stopping in 2s...
 *    Polly instance 'conn/save/3' stopping in 3s...
 *    Polly instance 'conn/4' stopping in 1s...
 *    Polly instance 'conn/save/3' stopping in 2s...
 *    Polly instance 'conn/4' stopped.
 *    Polly instance 'conn/save/3' stopping in 1s...
 *    Polly instance 'conn/save/3' stopped.
 *
 * 3. Validate your freshly recorded mock responses are complete and working.
 *    Re-run the exact command you want to test in mock mode (see step #1).
 *
 * 4. Write your test.
 *    Make sure to use the exact command including number of arguments and params.
 *
 * 5. Commit both your test and your new recordings to the repository.
 *    Your tests are likely going to reside outside the frodo-lib project but
 *    the recordings must be committed to the frodo-lib project.
 */

/*
To update cloud exports, run these:
rm test/e2e/exports/all/all.cloud.json
rm -rf test/e2e/exports/all-separate/cloud
FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config export -NRdaD test/e2e/exports/all -f all.cloud.json --include-active-values
FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config export -NRdxAD test/e2e/exports/all-separate/cloud --include-active-values

To update classic exports, ensure you have a local on-prem instance of AM with the host http://openam-frodo-dev.classic.com:8080/am, then run these:
rm test/e2e/exports/all/all.classic.json
rm -rf test/e2e/exports/all-separate/classic
FRODO_NO_CACHE=1 FRODO_HOST=http://openam-frodo-dev.classic.com:8080/am frodo config export -NRdaD test/e2e/exports/all -f all.classic.json --include-active-values
FRODO_NO_CACHE=1 FRODO_HOST=http://openam-frodo-dev.classic.com:8080/am frodo config export -NRdxAD test/e2e/exports/all-separate/classic --include-active-values

To record, run these:

// Cloud
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -adf test/e2e/exports/all/all.cloud.json
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import --all --clean --re-uuid-scripts --re-uuid-journeys --file test/e2e/exports/all/all.cloud.json
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -aCf test/e2e/exports/all/all.cloud.json
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -AD test/e2e/exports/all-separate/cloud
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import --all-separate --clean --re-uuid-scripts --re-uuid-journeys --directory test/e2e/exports/all-separate/cloud
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -CAD test/e2e/exports/all-separate/cloud
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import --default -CAD test/e2e/exports/all-separate/cloud
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -AD test/e2e/exports/all-separate/cloud --include-active-values
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -gf test/e2e/exports/all-separate/cloud/global/sync/sync.idm.json
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import --file test/e2e/exports/all-separate/cloud/realm/root-alpha/script/mode.script.json

FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -af test/e2e/exports/all/all.cloud.json --compare-and-delete --dry-run
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -af test/e2e/exports/all/all.cloud.json --compare-and-delete 
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -af test/e2e/exports/all/all.cloud.json --compare-and-delete --include-active-values

FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -AD test/e2e/exports/all-separate/cloud --compare-and-delete --dry-run
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -AD test/e2e/exports/all-separate/cloud --compare-and-delete 
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo config import -AD test/e2e/exports/all-separate/cloud --compare-and-delete --include-active-values

// Classic
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=http://openam-frodo-dev.classic.com:8080/am frodo config import -adf test/e2e/exports/all/all.classic.json -m classic
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=http://openam-frodo-dev.classic.com:8080/am frodo config import --all --clean --re-uuid-scripts --re-uuid-journeys --include-active-values --file test/e2e/exports/all/all.classic.json --type classic
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=http://openam-frodo-dev.classic.com:8080/am frodo config import -AdD test/e2e/exports/all-separate/classic -m classic
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=http://openam-frodo-dev.classic.com:8080/am frodo config import --all-separate --clean --re-uuid-scripts --re-uuid-journeys --include-active-values --directory test/e2e/exports/all-separate/classic --type classic
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=http://openam-frodo-dev.classic.com:8080/am frodo config import -gf test/e2e/exports/all-separate/classic/global/server/01.server.json -m classic
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=http://openam-frodo-dev.classic.com:8080/am frodo config import --global --file test/e2e/exports/all-separate/classic/global/authenticationModules/authPushReg.authenticationModules.json --type classic
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=http://openam-frodo-dev.classic.com:8080/am frodo config import -f test/e2e/exports/all-separate/classic/realm/root/webhookService/Cool-Webhook.webhookService.json -m classic
*/
import cp from 'child_process';
import { promisify } from 'util';
import { getEnv, removeAnsiEscapeCodes, removeProgressBarOutput } from './utils/TestUtils';
import { connection as c, classic_connection as cc } from './utils/TestConfig';

const exec = promisify(cp.exec);

process.env['FRODO_MOCK'] = '1';
const env = getEnv(c);
const classicEnv = getEnv(cc);

const allDirectory = 'test/e2e/exports/all';
const allCloudFileName = 'all.cloud.json';
const allClassicFileName = 'all.classic.json';
const allCloudExport = `${allDirectory}/${allCloudFileName}`;
const allClassicExport = `${allDirectory}/${allClassicFileName}`;
const allSeparateCloudDirectory = `test/e2e/exports/all-separate/cloud`;
const allSeparateClassicDirectory = `test/e2e/exports/all-separate/classic`;

describe('frodo config import', () => {
  test(`"frodo config import -adf ${allCloudExport}" Import everything from "${allCloudFileName}", including default scripts.`, async () => {
    const CMD = `frodo config import -adf ${allCloudExport}`;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch. 
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });

  // TODO: Fix test. Unable get test passing consistently, even after recording mocks (probably due to the re-uuid stuff). Skip for the meantime
  test.skip(`"frodo config import --all --clean --re-uuid-scripts --re-uuid-journeys --file ${allCloudExport}" Import everything from "${allCloudFileName}". Clean old services, and re-uuid journeys and scripts.`, async () => {
    const CMD = `frodo config import --all --clean --re-uuid-scripts --re-uuid-journeys --file ${allCloudExport}`;
    const { stdout } = await exec(CMD, env);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });

  test(`"frodo config import -aCf ${allCloudExport}" Import everything from "${allCloudFileName}". Clean old services`, async () => {
    const CMD = `frodo config import -aCf ${allCloudExport}`;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch. 
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });

  test(`"frodo config import -AD ${allSeparateCloudDirectory}" Import everything from directory "${allSeparateCloudDirectory}"`, async () => {
    const CMD = `frodo config import -AD ${allSeparateCloudDirectory}`;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch.
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });

  // TODO: Fix test. Unable get test passing consistently, even after recording mocks (probably due to the re-uuid stuff). Skip for the meantime
  test.skip(`"frodo config import --all-separate --clean --re-uuid-scripts --re-uuid-journeys --directory ${allSeparateCloudDirectory}" Import everything from directory "${allSeparateCloudDirectory}". Clean old services, and re-uuid journeys and scripts.`, async () => {
    const CMD = `frodo config import --all-separate --clean --re-uuid-scripts --re-uuid-journeys --directory ${allSeparateCloudDirectory}`;
    const { stdout } = await exec(CMD, env);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });

  test(`"frodo config import -CAD ${allSeparateCloudDirectory}" Import everything from directory "${allSeparateCloudDirectory}". Clean old services`, async () => {
    const CMD = `frodo config import -CAD ${allSeparateCloudDirectory}`;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch. 
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });

  test(`"frodo config import --default -CAD ${allSeparateCloudDirectory}" Import everything from directory "${allSeparateCloudDirectory}", including default scripts. Clean old services`, async () => {
    const CMD = `frodo config import --default -CAD ${allSeparateCloudDirectory}`;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch. 
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });

  test(`"frodo config import -AD ${allSeparateCloudDirectory} --include-active-values" Import everything with secret values from directory "${allSeparateCloudDirectory}"`, async () => {
    const CMD = `frodo config import -AD ${allSeparateCloudDirectory} --include-active-values`;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch. 
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });

  test(`"frodo config import -gf test/e2e/exports/all-separate/cloud/global/sync/sync.idm.json" Import sync.idm.json along with extracted mappings and no errors`, async () => {
    const CMD = `frodo config import -gf test/e2e/exports/all-separate/cloud/global/sync/sync.idm.json`;
    const { stdout } = await exec(CMD, env);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });

  test(`"frodo config import --file test/e2e/exports/all-separate/cloud/realm/root-alpha/script/mode.script.json" Import mode.script.json long with extracted scripts and no errors`, async () => {
    const CMD = `frodo config import --file test/e2e/exports/all-separate/cloud/realm/root-alpha/script/mode.script.json`;
    const { stdout } = await exec(CMD, env);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });

  test(`"frodo config import -adf ${allClassicExport} -m classic" Import everything from "${allClassicFileName}", including default scripts.`, async () => {
    const CMD = `frodo config import -adf ${allClassicExport} -m classic`;
    try {
      await exec(CMD, classicEnv);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch.
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  }, 300000);

  // TODO: Fix test. Unable get test passing consistently, even after recording mocks (probably due to the re-uuid stuff). Skip for the meantime
  test.skip(`"frodo config import --all --clean --re-uuid-scripts --re-uuid-journeys --include-active-values --file ${allClassicExport} --type classic" Import everything from "${allClassicFileName}". Clean old services, and re-uuid journeys and scripts.`, async () => {
    const CMD = `frodo config import --all --clean --re-uuid-scripts --re-uuid-journeys --include-active-values --file ${allClassicExport}--type classic`;
    const { stdout } = await exec(CMD, classicEnv);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });

  test(`"frodo config import -AdD ${allSeparateClassicDirectory} -m classic" Import everything from directory "${allSeparateClassicDirectory}"`, async () => {
    const CMD = `frodo config import -AdD ${allSeparateClassicDirectory} -m classic`;
    try {
      await exec(CMD, classicEnv);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch.
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  }, 300000);

  // TODO: Fix test. Unable get test passing consistently, even after recording mocks (probably due to the re-uuid stuff). Skip for the meantime
  test.skip(`"frodo config import --all-separate --clean --re-uuid-scripts --re-uuid-journeys --include-active-values --directory ${allSeparateClassicDirectory} --type classic" Import everything from directory "${allSeparateClassicDirectory}". Clean old services, and re-uuid journeys and scripts.`, async () => {
    const CMD = `frodo config import --all-separate --clean --re-uuid-scripts --re-uuid-journeys --include-active-values --directory ${allSeparateClassicDirectory} --type classic`;
    const { stdout } = await exec(CMD, classicEnv);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });

  test(`"frodo config import -gf test/e2e/exports/all-separate/classic/global/server/01.server.json -m classic" Import server 01 along with extracted properties and no errors`, async () => {
    const CMD = `frodo config import -gf test/e2e/exports/all-separate/classic/global/server/01.server.json -m classic`;
    const { stdout } = await exec(CMD, classicEnv);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });

  test(`"frodo config import --global --file test/e2e/exports/all-separate/classic/global/authenticationModules/authPushReg.authenticationModules.json --type classic" Fail to import authentication module due to it being read only.`, async () => {
    const CMD = `frodo config import --global --file test/e2e/exports/all-separate/classic/global/authenticationModules/authPushReg.authenticationModules.json --type classic`;
    try {
      await exec(CMD, classicEnv);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch.
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });

  test(`"frodo config import -f test/e2e/exports/all-separate/classic/realm/root/webhookService/Cool-Webhook.webhookService.json -m classic" Import the webhook service with no errors`, async () => {
    const CMD = `frodo config import -f test/e2e/exports/all-separate/classic/realm/root/webhookService/Cool-Webhook.webhookService.json -m classic`;
    const { stdout } = await exec(CMD, classicEnv);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });


  test(`"frodo config import -af ${allCloudExport} --compare-and-delete --dry-run" Import the webhook service with no errors`, async () => {
    const CMD = `frodo config import af ${allCloudExport} --compare-and-delete --dry-run`;
    const { stdout } = await exec(CMD, env);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });

  test(`"frodo config import -af ${allCloudExport} --compare-and-delete" Import everything from "${allCloudExport}", including default scripts.`, async () => {
    const CMD = `frodo config import -af ${allCloudExport} --compare-and-delete `;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch.
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  }, 300000);

  test(`"frodo config import -af ${allCloudExport} --compare-and-delete --include-active-values" Import everything from "${allCloudExport}", including default scripts.`, async () => {
    const CMD = `frodo config import -af ${allCloudExport} --compare-and-delete --include-active-values`;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch.
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });


  test(`"frodo config import -AD ${allSeparateCloudDirectory} --compare-and-delete --include-active-values --dry-run" Import everything with secret values from directory "${allSeparateCloudDirectory}"`, async () => {
    const CMD = `frodo config import -AD ${allSeparateCloudDirectory} --compare-and-delete --dry-run openam`;
    const { stdout } = await exec(CMD, env);
    expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
  });

  test(`"frodo config import -AD ${allSeparateCloudDirectory} compare-and-delete" Import everything with secret values from directory "${allSeparateCloudDirectory}"`, async () => {
    const CMD = `frodo config import -AD ${allSeparateCloudDirectory} --compare-and-delete`;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch. 
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });
 
  test(`"frodo config import -AD ${allSeparateCloudDirectory} --compare-and-delete --include-active-values" Import everything with secret values from directory "${allSeparateCloudDirectory}"`, async () => {
    const CMD = `frodo config import -AD ${allSeparateCloudDirectory} --compare-and-delete --include-active-values`;
    try {
      await exec(CMD, env);
      fail("Command should've failed")
    } catch (e) {
      // parallel test execution alters the progress bar output causing the snapshot to mismatch. 
      // only workable solution I could find was to remove progress bar output altogether from such tests.
      expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
    }
  });

});