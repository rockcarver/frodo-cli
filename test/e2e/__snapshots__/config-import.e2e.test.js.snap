// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frodo config import "frodo config import --default -CAD test/e2e/exports/all-separate/cloud" Import everything from directory "test/e2e/exports/all-separate/cloud", including default scripts. Clean old services 1`] = `
"Connected to https://openam-frodo-dev.forgeblocks.com/am [alpha] as service account Frodo-SA-1720799681233 [b672336b-41ef-428d-ae4a-e0c082875377]
Error updating config entity endpoint/Test
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
Error updating config entity endpoint/testEndpoint2
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
Error Importing Services
  Error importing services
  Error putting global full service configs
  Error putting global full service config CorsService
  Error deleting service 'CorsService' before import
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: The singleton resource cannot be deleted
  Error putting global full service config dashboard
  Error deleting service 'dashboard' before import
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: The singleton resource cannot be deleted
Error updating script
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 403
    Reason: Forbidden
    Message: This operation is not available in PingOne Advanced Identity Cloud.
Error updating script
  HTTP client error
    Code: ERR_BAD_RESPONSE
    Status: 502
- Resolving dependencies
✔ Resolved all dependencies.
Error Importing Services
  Error importing services
  Error putting realm full service configs
  Error putting realm full service config email
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: Data validation failed for the attribute, Transport Type
Error updating script
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 403
    Reason: Forbidden
    Message: This operation is not available in PingOne Advanced Identity Cloud.
Error updating script
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 403
    Reason: Forbidden
    Message: This operation is not available in PingOne Advanced Identity Cloud.
- Resolving dependencies
✔ Resolved all dependencies.
Error Importing Services
  Error importing services
  Error putting realm full service configs
  Error putting realm full service config TNTPPingOneService
  Error putting descendent 'JC' of service 'TNTPPingOneService'
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: Invalid config: Unable to add subConfig PingOneService
"
`;

exports[`frodo config import "frodo config import --file test/e2e/exports/all-separate/cloud/realm/root-alpha/script/mode.script.json" Import mode.script.json long with extracted scripts and no errors 1`] = `""`;

exports[`frodo config import "frodo config import --global --file test/e2e/exports/all-separate/classic/global/authenticationModules/authPushReg.authenticationModules.json --type classic" Fail to import authentication module due to it being read only. 1`] = `
"Connected to http://openam-frodo-dev.classic.com:8080/am [/] as user amAdmin
No imports were made
"
`;

exports[`frodo config import "frodo config import -AD test/e2e/exports/all-separate/cloud --include-active-values" Import everything with secret values from directory "test/e2e/exports/all-separate/cloud" 1`] = `
"Connected to https://openam-frodo-dev.forgeblocks.com/am [alpha] as service account Frodo-SA-1720799681233 [b672336b-41ef-428d-ae4a-e0c082875377]
Error updating config entity endpoint/Test
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
Error updating config entity endpoint/testEndpoint2
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
- Resolving dependencies
✔ Resolved all dependencies.
Error Importing Services
  Error importing services
  Error putting realm full service configs
  Error putting realm full service config email
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: Data validation failed for the attribute, Transport Type
- Resolving dependencies
✔ Resolved all dependencies.
"
`;

exports[`frodo config import "frodo config import -AD test/e2e/exports/all-separate/cloud" Import everything from directory "test/e2e/exports/all-separate/cloud" 1`] = `
"Connected to https://openam-frodo-dev.forgeblocks.com/am [alpha] as service account Frodo-SA-1720799681233 [b672336b-41ef-428d-ae4a-e0c082875377]
Error updating config entity endpoint/Test
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
- Resolving dependencies
✔ Resolved all dependencies.
Error Importing Services
  Error importing services
  Error putting realm full service configs
  Error putting realm full service config email
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: Data validation failed for the attribute, Transport Type
- Resolving dependencies
✔ Resolved all dependencies.
"
`;

exports[`frodo config import "frodo config import -AdD test/e2e/exports/all-separate/classic -m classic" Import everything from directory "test/e2e/exports/all-separate/classic" 1`] = `
"Connected to http://openam-frodo-dev.classic.com:8080/am [/] as user amAdmin
Error Importing Authentication Settings
  Error importing authentication settings
  Error updating authentication settings
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 404
    Reason: Not Found
    Message: Resource '' not found
- Resolving dependencies
✔ Resolved all dependencies.
- Resolving dependencies
✔ Resolved all dependencies.
- Resolving dependencies
✔ Resolved all dependencies.
"
`;

exports[`frodo config import "frodo config import -CAD test/e2e/exports/all-separate/cloud" Import everything from directory "test/e2e/exports/all-separate/cloud". Clean old services 1`] = `
"Connected to https://openam-frodo-dev.forgeblocks.com/am [alpha] as service account Frodo-SA-1720799681233 [b672336b-41ef-428d-ae4a-e0c082875377]
Error updating config entity endpoint/Test
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
Error Importing Services
  Error importing services
  Error putting global full service configs
  Error putting global full service config CorsService
  Error deleting service 'CorsService' before import
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: The singleton resource cannot be deleted
  Error putting global full service config dashboard
  Error deleting service 'dashboard' before import
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: The singleton resource cannot be deleted
- Resolving dependencies
✔ Resolved all dependencies.
Error Importing Services
  Error importing services
  Error putting realm full service configs
  Error putting realm full service config email
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: Data validation failed for the attribute, Transport Type
- Resolving dependencies
✔ Resolved all dependencies.
"
`;

exports[`frodo config import "frodo config import -aCf test/e2e/exports/all/all.cloud.json" Import everything from "all.cloud.json". Clean old services 1`] = `
"Connected to https://openam-frodo-dev.forgeblocks.com/am [alpha] as service account Frodo-SA-1720799681233 [b672336b-41ef-428d-ae4a-e0c082875377]
Error updating config entity endpoint/Test
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
Error updating config entity endpoint/testEndpoint2
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
Error Importing Services
  Error importing services
  Error putting global full service configs
  Error putting global full service config CorsService
  Error deleting service 'CorsService' before import
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: The singleton resource cannot be deleted
  Error putting global full service config dashboard
  Error deleting service 'dashboard' before import
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: The singleton resource cannot be deleted
- Resolving dependencies
✔ Resolved all dependencies.
Error Importing Services
  Error importing services
  Error putting realm full service configs
  Error putting realm full service config email
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: Data validation failed for the attribute, Transport Type
- Resolving dependencies
✔ Resolved all dependencies.
"
`;

exports[`frodo config import "frodo config import -adf test/e2e/exports/all/all.classic.json -m classic" Import everything from "all.classic.json", including default scripts. 1`] = `
"Connected to http://openam-frodo-dev.classic.com:8080/am [/] as user amAdmin
Error Importing Authentication Settings
  Error importing authentication settings
  Error updating authentication settings
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 404
    Reason: Not Found
    Message: Resource '' not found
- Resolving dependencies
✔ Resolved all dependencies.
- Resolving dependencies
✔ Resolved all dependencies.
- Resolving dependencies
✔ Resolved all dependencies.
"
`;

exports[`frodo config import "frodo config import -adf test/e2e/exports/all/all.cloud.json" Import everything from "all.cloud.json", including default scripts. 1`] = `
"Connected to https://openam-frodo-dev.forgeblocks.com/am [alpha] as service account Frodo-SA-1720799681233 [b672336b-41ef-428d-ae4a-e0c082875377]
Error updating config entity endpoint/Test
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
Error updating config entity endpoint/testEndpoint2
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 401
    Reason: Unauthorized
    Message: Access Denied
Error updating script
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 403
    Reason: Forbidden
    Message: This operation is not available in PingOne Advanced Identity Cloud.
Error updating script
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 403
    Reason: Forbidden
    Message: This operation is not available in PingOne Advanced Identity Cloud.
- Resolving dependencies
✔ Resolved all dependencies.
Error Importing Services
  Error importing services
  Error putting realm full service configs
  Error putting realm full service config email
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 400
    Reason: Bad Request
    Message: Data validation failed for the attribute, Transport Type
Error updating script
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 403
    Reason: Forbidden
    Message: This operation is not available in PingOne Advanced Identity Cloud.
Error updating script
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 403
    Reason: Forbidden
    Message: This operation is not available in PingOne Advanced Identity Cloud.
- Resolving dependencies
✔ Resolved all dependencies.
"
`;

exports[`frodo config import "frodo config import -f test/e2e/exports/all-separate/classic/realm/root/webhookService/Cool-Webhook.webhookService.json -m classic" Import the webhook service with no errors 1`] = `""`;

exports[`frodo config import "frodo config import -gf test/e2e/exports/all-separate/classic/global/server/01.server.json -m classic" Import server 01 along with extracted properties and no errors 1`] = `""`;

exports[`frodo config import "frodo config import -gf test/e2e/exports/all-separate/cloud/global/sync/sync.idm.json" Import sync.idm.json along with extracted mappings and no errors 1`] = `""`;
